### PR TITLE
Adjust foot forces and grav comp using current orientation.

### PR DESCRIPTION
**Pull request by iamtesch**

### CODE REVIEW

File: kits/daisy/src/robot/hexapod.cpp [hunk 1]
```diff
@@ -501,16 +500,19 @@ Hexapod::Hexapod(std::shared_ptr<Group> group,
   last_step_legs_.insert(3);
   last_step_legs_.insert(4);
 
+  // Default to straight down w/ a level chassis
+  gravity_direction_ << 0, 0, -1;
```
Comment (rLinks234) [inline kits/daisy/src/robot/hexapod.cpp]: change to `gravity_direction_ = -Eigen::Vector3d::UnitZ();`

File: kits/daisy/src/robot/hexapod.cpp [hunk 2]
```diff
@@ -533,25 +535,26 @@ Hexapod::Hexapod(std::shared_ptr<Group> group,
...
-      // 1) there are other factors we don't have in there yet (e.g., grav comp)
-      // 2) the jacobian is slightly noisy
-      // Note that the foot forces have been compared to MATLAB results, and seem
-      // reasonable. No further debugging has been done.
-      avg_accel /= num_legs_used * 2.0;
-      avg_gyro /= num_legs_used * 2.0;
+      avg_grav /= num_legs_used;
+      // TODO: ideally, put a lock around this...but this is the only place
+      // we write to it, so it isn't terrible.
+      gravity_direction_ = avg_grav;
```
Comment (rLinks234) [inline kits/daisy/src/robot/hexapod.cpp]: **Explicitly comment at declaration of this variable that it is not thread safe**

File: kits/daisy/src/robot/hexapod.hpp [hunk 3]
```diff
@@ -91,6 +91,8 @@ class Hexapod
   void clearLegColors();
   void setLegColor(int leg_index, uint8_t r, uint8_t g, uint8_t b);
 
+  Eigen::Vector3d getGravityDirection() const { return gravity_direction_; }
```
Comment (rLinks234) [inline kits/daisy/src/robot/hexapod.hpp]: return const reference

File: kits/daisy/src/robot/hexapod.hpp [hunk 4]
```diff
@@ -121,6 +123,9 @@ class Hexapod
 
   Eigen::Vector3d vel_xyz_;
 
+  // The orientation of gravity, as a unit vector, w.r.t. the chassis.
+  Eigen::Vector3d gravity_direction_;
```
Comment (rLinks234) [inline kits/daisy/src/robot/hexapod.hpp]: **Explicitly comment that this variable is not accessed or mutated in a thread safe manner**

**Comment by rLinks234** [comment]
For most instances, lock contention is not going to be an issue. We're in an unmanaged language, where a VM isn't controlling the execution of our code, which means we don't have to worry as much about mutexes as languages like Java or C#. Java and C# require explicit synchronization when threads leave and enter a VM context.

Lock contention will only start to happen when *the lock is held and released tens of thousands of times a second by multiple threads* when we start to reach a few magnitudes close to the range of how long system calls take (plus a few `ns` to a few dozen `ns` for the implementation of `futex_wait`)

File: kits/daisy/src/robot/hexapod.hpp (hunk 3 above)
Comment (iamtesch) [reply kits/daisy/src/robot/hexapod.hpp]: I'd prefer to return a copy of the variable, so any thread-safety issues are isolated to this class.  Is that reasonable?  If it was a heavyweight class, I'd be more concerned.

Comment (rLinks234) [reply kits/daisy/src/robot/hexapod.hpp]: If any issues arise, it's because we haven't added enough documentation that it is not thread safe

**Pull request merged**